homeassistant/components/velbus/__init__.py

```diff
@@ -80,10 +79,18 @@ def callback():
                 hass.config_entries.async_forward_entry_setup(
                     entry, category))
 
-    controller.scan(callback)
+    try:
+        controller = velbus.Controller(entry.data[CONF_PORT])
+        controller.scan(callback)
+    except Exception as err:   # pylint: disable=broad-except
```

Comment:
What error are we expecting. We shouldn't catch broad expectations unless absolutely necessary. Usually we can predict what errors are possible.

Reply:
The problem is that the python-velbus lib does not do any catching so i already identified at least 6 possible exceptions that could be thrown ....
This is the reason i used a the general Exception

Reply:
What are the exceptions?

Reply:
below is a list i remember
- serial.serialutil.SerialException
- socket.error
- VelbusException
- TypeError (comming from the socket lib)
- RuntimeError  (from the threading lib)

Reply:
The velbus library does catch some exceptions, eg here:
https://github.com/thomasdelaet/python-velbus/blob/dc1ac6bde6215ce668c1cced7e527e5c070ac6d1/velbus/connections/serial.py#L54-L57

We should catch `VelbusException` as needed here. The serial and maybe socket exceptions should be caught in the library and handled similarly as above, if the user of the library needs to handle them. The other exceptions looks more like symptoms of bugs and should be fixed in the library.

Please address the remaining exceptions in the library repo.